mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -145,13 +135,20 @@ class LogisticRegressionModel (
        */
       var bestClass = 0
       var maxMargin = 0.0
-      var i = 0
-      while(i < margins.size) {
-        if (margins(i) > maxMargin) {
-          maxMargin = margins(i)
+      val withBias = if (dataMatrix.size + 1 == dataWithBiasSize) true else false
```

Comment:
Ah, no, my point is much simpler, just saying that `if (x) true else false` is equivalent to the simpler expression `x`. I'm not questioning the boolean expression itself.